apps/user-ms/src/main/java/it/pagopa/selfcare/user/util/UserUtils.java

```diff
@@ -86,12 +87,13 @@ public List<UserNotificationToSend> buildUsersNotificationResponse(UserInstituti
                 .toList();
     }
 
-    public UserNotificationToSend buildUserNotificationToSend(UserInstitution userInstitution, UserResource userResource, String productId, OnboardedProductState status) {
+    public UserNotificationToSend buildUserNotificationToSend(UserInstitution userInstitution, UserResource userResource, String productId, String productRole, OnboardedProductState status) {
         UserToNotify userToNotify = buildUserToNotify(userResource, userInstitution, status);
         UserNotificationToSend userNotificationToSend = buildUserNotificationToSend(userInstitution, productId);
 
         userInstitution.getProducts().stream()
-                .filter(p -> org.apache.commons.lang3.StringUtils.equals(p.getProductId(), productId))
+                .filter(p -> org.apache.commons.lang3.StringUtils.equals(p.getProductId(), productId)
+                 && (org.apache.commons.lang3.StringUtils.isBlank(productRole) || org.apache.commons.lang3.StringUtils.equals(p.getProductRole(), productRole)))
```

Comment:
could you create a reusable method and using that in other points, for ex. buildCreateEmailDataModel